integration_tests/waypoints.cpp

```diff
@@ -64,276 +37,171 @@ TEST_F(SitlTest, MissionAddWaypointsAndFly)
-                break;
-            case MissionState::MISSION:
-                // Pause after 15s
-                std::this_thread::sleep_for(std::chrono::seconds(15));
-                std::cout << "Pause mission again" << std::endl;
-                device.mission().pause_mission_async(
-                    std::bind(&receive_pause_mission_result, _1));
-                _mission_state = MissionState::MISSION_PAUSING;
-                break;
-            case MissionState::MISSION_PAUSING:
-                break;
-            case MissionState::PAUSE:
-                // pause 2s, then carry on
-                std::this_thread::sleep_for(std::chrono::seconds(2));
-                std::cout << "Start mission again" << std::endl;
-                device.mission().start_mission_async(
-                    std::bind(&receive_continue_mission_result, _1));
-                _mission_state = MissionState::MISSION_RESUMING;
-                break;
-            case MissionState::MISSION_RESUMING:
-
-                if (!_jump_done) {
-                    // Jump back to previous mission item
-                    std::cout << "Send current mission item" << std::endl;
-                    device.mission().set_current_mission_item_async(
-                        3,
-                        std::bind(&receive_set_current_mission_item_result, _1));
-                    _jump_done = true;
-                }
-                break;
-            case MissionState::MISSION_DONE:
-                device.action().return_to_launch_async(
-                    std::bind(&receive_return_to_launch_result, _1));
-                _mission_state = MissionState::RETURN;
-                break;
-            case MissionState::RETURN:
-                if (!device.telemetry().in_air()) {
-                    _mission_state = MissionState::RETURN_DONE;
-                }
-                break;
-            case MissionState::RETURN_DONE:
-                device.action().disarm_async(std::bind(&receive_disarm_result, _1));
-                break;
-            case MissionState::DONE:
-                finished = true;
-                break;
-
-            case MissionState::ERROR:
-                finished = true;
-                break;
-
-        }
-
-        std::this_thread::sleep_for(std::chrono::milliseconds(10));
+        add_mission_item(47.398001890458097,
+                         8.5455576181411743,
+                         10.0f, 5.0f, false,
+                         0.0f, 0.0f,
+                         MissionItem::CameraAction::STOP_PHOTO_INTERVAL));
+
+    {
+        // We only have the send_mission function asynchronous for now, so we wrap it using
+        // std::future.
+        auto prom = std::make_shared<std::promise<void>>();
+        auto future_result = prom->get_future();
+        device.mission().send_mission_async(
+        mission_items, [prom](Mission::Result result) {
+            EXPECT_EQ(result, Mission::Result::SUCCESS);
```

Comment:
What happens if they aren't equal inside async request? Does this propagate out of the promise? 
For test code I guess you want this to exit/crash but I'd hope for example code we'd get sensible handling.

Reply:
You are right that we should crash because there is no point in continuing if that fails.

For the integration_test what I we should do is make it an `ASSERT_EQ` which exits the test if it fails.

For the example we can either `abort()`, or propagate the result out of the promise and check for it.